pkg/index/index.go

```diff
@@ -78,17 +80,25 @@ func (c *State) UpsertWorkspace(shard string, ws *tenancyv1alpha1.Workspace) {
 	clusterName := logicalcluster.From(ws)
 
 	c.lock.RLock()
-	got := c.shardWorkspaceNameCluster[shard][clusterName][ws.Name]
+	cluster := c.shardWorkspaceNameCluster[shard][clusterName][ws.Name]
+	mount := c.workspaceNameMount[clusterName][ws.Name] // experimental feature
```

Comment:
what is a "mount"? A URL? A name?

Reply:
At this point, I'm thinking maybe we should store just a URL, not a full object. Good question.

Reply:
Sounds fine. Maybe call it mountURL ?

Reply:
hmmm. It's a bit more complicated when I read it again. I stored full serialized objects so I could do "poor man DeepCopy". Meaning it would update this in the cache if anything changes in the mount objects. And idea behind this is that we will need to make a workspace not ready if a mountpoint becomes not ready. 

And due to this, I stored all serialized objects. As alternative is to store some aggregate of status.

Plus as we store in serialized to the string in the annotations, saves unmarshaling to objects here.